Bugzilla 275 XIP: - while the kernel is booting it crashes in the NAND driver
Applied TLSbo79471_fix_XIP_Kernel_crash_with_NAND_driver_on_IMX31ADS.patch

cvs diff -puN -r1.2 -r1.3 drivers/mtd/nand/Kconfig
Index: drivers/mtd/nand/Kconfig
===================================================================
RCS file: drivers/mtd/nand/Kconfig,v
retrieving revision 1.2
retrieving revision 1.3
diff -p -u -r1.2 -r1.3
--- linux-2.6.19.2-mx/drivers/mtd/nand/Kconfig	15 Jan 2007 05:44:49 -0000	1.2
+++ linux-2.6.19.2-mx/drivers/mtd/nand/Kconfig	16 Jan 2007 23:57:33 -0000	1.3
@@ -11,6 +11,8 @@ config MTD_NAND
 	help
 	  This enables support for accessing all type of NAND flash
 	  devices. For further information see
+	  MXC NAND driver cannot be compiled for MX31ADS with XIP enabled.
+	  Refer Errata on EMI for using NOR (WEIM) and NAND simultaneously 
 	  <http://www.linux-mtd.infradead.org/doc/nand.html>.
 
 config MTD_NAND_VERIFY_WRITE
@@ -242,7 +244,7 @@ config MTD_NAND_NANDSIM
 config MTD_NAND_MXC
 	tristate "MXC NAND support"
 	depends on MTD_NAND
-	depends on ARCH_MXC
+	depends on ARCH_MXC && (!MACH_MX31ADS || !XIP_KERNEL)
 	help
 	  This enables the driver for the NAND flash controller on the
 	  MXC processors.
